### PR TITLE
Add RBAC support

### PR DESCRIPTION
**Pull request by mcinquin**

### CODE REVIEW

**Comment by mcinquin** [comment]
Up ?

**Comment by abhishekray07** [comment]
Could you investigate why the build is failing?

File: airflow_prometheus_exporter/prometheus_exporter.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 from airflow.models import DagModel, DagRun, TaskInstance, TaskFail
 from airflow.plugins_manager import AirflowPlugin
 from airflow.settings import Session
+from airflow.settings import RBAC
```
Comment (abhishekray07) [inline airflow_prometheus_exporter/prometheus_exporter.py]: Better to write as `from airflow.settings import RBAC, Session`

**Review by abhishekray07 — CHANGES_REQUESTED** [review]
Requesting changes for minor code organization

**Comment by mcinquin** [comment]
> Could you investigate why the build is failing?

 It's due to a version problem between flask-appbuilder and marshmallow. 
```
ERROR: flask-appbuilder 1.13.1 has requirement marshmallow<2.20,>=2.18.0, but you'll have marshmallow 3.2.2 which is incompatible.
```

On my automation script, i fixed the problem by splitting airflow and airflow-prometheus-exporter installation:

1. pip3 install airflow
2. pip3 install airflow-prometheus-exporter

**Comment by mcinquin** [comment]
The code is fully operational with Airflow configured with RBAC.  But there is a python Traceback when airflow is configured without RBAC.
My skills are not sufficient to fix the problem. Any help would be welcome.

```
Nov 28 14:10:30 airflow-1 python3[10974]: Traceback (most recent call last):
Nov 28 14:10:30 airflow-1 python3[10974]:   File "/home/airflow/airflow/venv/bin/airflow", line 37, in <module>
Nov 28 14:10:30 airflow-1 python3[10974]:     args.func(args)
Nov 28 14:10:30 airflow-1 python3[10974]:   File "/home/airflow/airflow/venv/lib/python3.5/site-packages/airflow/utils/cli.py", line 74, in wrapper
Nov 28 14:10:30 airflow-1 python3[10974]:     return f(*args, **kwargs)
Nov 28 14:10:30 airflow-1 python3[10974]:   File "/home/airflow/airflow/venv/lib/python3.5/site-packages/airflow/bin/cli.py", line 902, in webserver
Nov 28 14:10:30 airflow-1 python3[10974]:     app = cached_app_rbac(None) if settings.RBAC else cached_app(None)
Nov 28 14:10:30 airflow-1 python3[10974]:   File "/home/airflow/airflow/venv/lib/python3.5/site-packages/airflow/www/app.py", line 199, in cached_app
Nov 28 14:10:30 airflow-1 python3[10974]:     app = create_app(config, testing)
Nov 28 14:10:30 airflow-1 python3[10974]:   File "/home/airflow/airflow/venv/lib/python3.5/site-packages/airflow/www/app.py", line 160, in create_app
Nov 28 14:10:30 airflow-1 python3[10974]:     integrate_plugins()
Nov 28 14:10:30 airflow-1 python3[10974]:   File "/home/airflow/airflow/venv/lib/python3.5/site-packages/airflow/www/app.py", line 151, in integrate_plugins
Nov 28 14:10:30 airflow-1 python3[10974]:     log.debug('Adding view %s', v.name)
Nov 28 14:10:30 airflow-1 python3[10974]: AttributeError: 'list' object has no attribute 'name'

```

File: docker/Dockerfile [hunk 2]
```diff
@@ -11,6 +11,7 @@ RUN set -ex \
         apt-utils \
     && useradd -ms /bin/bash -d ${AIRFLOW_USER_HOME} airflow \
     && pip install -U pip
+    && pip install apache-airflow
```
Comment (abhishekray07) [inline docker/Dockerfile]: why do we need this?
when we install airflow-prometheus-exporter on line 20, we are already installing apache-airflow?

**Review by abhishekray07 — CHANGES_REQUESTED** [review]

File: docker/Dockerfile (hunk 2 above)
Comment (abhishekray07) [reply docker/Dockerfile]: Do we need a newer version of Airflow to deal with this error? If so, you can update the version in this file: https://github.com/robinhood/airflow-prometheus-exporter/blob/master/setup.py#L12

Comment (mcinquin) [reply docker/Dockerfile]: As explained above, there is an issue with version compatibility between marshmallow and flask-appbuilder if we don't install apache-airflow before airflow-prometheus-exporter.

It's the only way I found as workaround. 

Honestly, I'm not going to investigate that any further.

Comment (abhishekray07) [reply docker/Dockerfile]: I am not sure what build issue you were seeing earlier but I am not seeing any issues if I run the build locally (without this change). If you want, you can try to remove this line and try again.

Comment (mcinquin) [reply docker/Dockerfile]: Without this change:
`ERROR: flask-appbuilder 1.13.1 has requirement marshmallow<2.20,>=2.18.0, but you'll have marshmallow 3.3.0 which is incompatible.`

And now you can see the result in travis.

Comment (svazquezco) [reply docker/Dockerfile]: Hello,

I'm interesting in this feature. Could I help with anything to unlock this PR?

Regards

**Pull request closed without merging**